### DISCUSSION TITLE
NREL Turbine Blade case failing in Tutorials

### DISCUSSION THREAD

**Discussion by justAPhDStudent** [mdolab/dafoam#78] [Tutorials]
Hello,
I tried recently a tutorial example for the NREL turbine blade case on 4 processors (laptop) with docker version of dafoam. The solution ran for almost 10K sec then failed with following output:
| pyDAFoam Error: Primal solution failed for the baseline design!            |
+------------------------------------------------------------------------------+

capi_return is NULL
Call-back cb_slfunc_in_slsqp__user__routines failed.
/usr/lib/python3.6/importlib/_bootstrap.py:219: RuntimeWarning: numpy.ufunc size changed, may indicate binary incompatibility. Expected 192 from C header, got 216 from PyObject
  return f(*args, **kwds)
/usr/lib/python3.6/importlib/_bootstrap.py:219: RuntimeWarning: numpy.ufunc size changed, may indicate binary incompatibility. Expected 192 from C header, got 216 from PyObject
  return f(*args, **kwds)
/usr/lib/python3.6/importlib/_bootstrap.py:219: RuntimeWarning: numpy.ufunc size changed, may indicate binary incompatibility. Expected 192 from C header, got 216 from PyObject
  return f(*args, **kwds)
/home/dafoamuser/.local/lib/python3.6/site-packages/idwarp/UnstructuredMesh.py:112: ResourceWarning: unclosed file <_io.BufferedReader name='/home/dafoamuser/.local/lib/python3.6/site-packages/idwarp/idwarp.so'>
  self.warp = MExt('idwarp', [curDir], debug=debug)._module
Traceback (most recent call last):
  File "runScript.py", line 164, in <module>
    sol = opt(optProb, sens=optFuncs.calcObjFuncSens, storeHistory=histFile)
  File "/home/dafoamuser/.local/lib/python3.6/site-packages/pyoptsparse/pySLSQP/pySLSQP.py", line 224, in __call__
    ngrad, slfunc, slgrad)
  File "/home/dafoamuser/.local/lib/python3.6/site-packages/pyoptsparse/pySLSQP/pySLSQP.py", line 174, in slfunc
    fobj, fcon, fail = self._masterFunc(x, ["fobj", "fcon"])
  File "/home/dafoamuser/.local/lib/python3.6/site-packages/pyoptsparse/pyOpt_optimizer.py", line 287, in _masterFunc
    result = self._masterFunc2(*args)
  File "/home/dafoamuser/.local/lib/python3.6/site-packages/pyoptsparse/pyOpt_optimizer.py", line 318, in _masterFunc2
    args = self.optProb.objFun(xuser)
  File "/home/dafoamuser/.local/lib/python3.6/site-packages/dafoam/optFuncs.py", line 54, in calcObjFuncValues
    DASolver.evalFunctions(funcs, evalFuncs=evalFuncs)
  File "/home/dafoamuser/.local/lib/python3.6/site-packages/dafoam/pyDAFoam.py", line 896, in evalFunctions
    raise Error("Primal solution failed for the baseline design!")
dafoam.pyDAFoam.Error

No modification have been done on the runscript in the tutorial.
With many thanks,
T.

**Comment by friedenhe** [comment]
Thanks for the feedback. The NREL6 tutorial was somehow not up-to-date. We have just fixed this issue. Note that you won't be able to run this case (800K cells) on your personal computer. We recommend running it on an HPC system with 80 CPU cores. Check this page for the [recommended resources](https://dafoam.github.io/mydoc_get_started_faq.html#how-many-cpu-cores-to-use-and-how-much-memory-does-it-need). Or, you have to reduce the size of the mesh to less than 100 K.

**Reply by justAPhDStudent** [reply]
Thanks very much for your feedback. Sorry that I forgot to mention, the problem was run on 50GB RAM (updated swap with SSD drive on laptop). I don't think there was a memory error; as I was monitoring that; and it was consuming around 30-35GB (in > $htop). 
Anyway. thanks again; and will try once the fix will be available. 
very happy to have found this wonderful platform.
Regards

**Reply by friedenhe** [reply]
To run it with 50 GB RAM, I suggest you change `hex ( 0 1 3 2 4 5 7 6) (10 10 10) simpleGrading (1 1 1)` to `hex ( 0 1 3 2 4 5 7 6) (6 6 6) simpleGrading (1 1 1)` in system/blockMeshDict. This should give you about 200K cells. The default setup has 800 K cells now and will eat all your memory. The 30-35 GB memory usage you saw was just for the flow solution. When the adjoint kicks in, it will use a lot of memory.
